options/rtdl/generic/main.cpp

```diff
@@ -335,65 +328,68 @@ void *__dlapi_get_tls(struct __abi_tls_entry *entry) {
 #ifdef __MLIBC_POSIX_OPTION
 
 extern "C" [[ gnu::visibility("default") ]]
-void *__dlapi_open(const char *file, int local, void *returnAddress) {
-	// TODO: Thread-safety!
-	auto rts = rtsCounter++;
+void *__dlapi_open(const char *file, int flags, void *returnAddress) {
+	if (logDlCalls)
+		mlibc::infoLogger() << "rtdl: __dlapi_open(\"" << (file ? file : "nullptr") << "\")" << frg::endlog;
+
+	auto unhandled = RTLD_NODELETE | RTLD_DEEPBIND;
+	if (flags & unhandled) {
+		mlibc::infoLogger() << "rtdl: dlopen flag " << (flags & unhandled)
+			<< "is unsupported" << frg::endlog;
+	}
 
 	if(!file)
 		return executableSO;
 
-	if(local)
-		mlibc::infoLogger() << "\e[31mrtdl: RTLD_LOCAL " << file << " is not supported properly\e[39m"
-				<< frg::endlog;
+	// TODO: Thread-safety!
+	auto rts = rtsCounter++;
+
+	// To know which RPATH / RUNPATH to process, we need to know which object called us.
+	auto findOrigin = [&](){
```

Comment:
Why does this need to be a lambda? It also doesn't seem to be used in all of the relevant cases.